### ISSUE TITLE
wrong __all__

### ISSUE THREAD

**Issue by seii-saintway** [fastai/nbdev#447]
```
# default_exp test
```
```
# export
def test():
    a = """
b = 0
c = 0
"""
```
↓
```
# AUTOGENERATED! DO NOT EDIT! File to edit: notebooks/test.ipynb (unless otherwise specified).

__all__ = ['test', 'b', 'c']

# Cell
def test():
    a = """
b = 0
c = 0
"""
```

The `__all__` is wrong.

**Comment by hamelsmu** [comment]
Why is it wrong?  Please explain.

**Comment by hamelsmu** [comment]
Since this is behaving as documented I'll go ahead and close the issue.  There is a flag you can place on the cell to prevent an export

**Comment by hamelsmu** [comment]
Note that nbdev doesn't exist in isolation.  We rely on this behavior for building the fastai libraries 

You can read about these flags here https://nbdev.fast.ai/export.html

**Comment by seii-saintway** [comment]
Thanks. I know I may use #exporti and #add2all if I want to add `test` to
`__all__`, although this way is not that elegant. And I may use `show_doc`
if I want to show the document which is hidden by #exporti before. I think
the features of this flag #exporti is not simple enough.
And sometimes I found that the notebook itself is more readable than the
document generated from it.

**Comment by seii-saintway** [comment]
Thanks.
I know I may use `#exporti` and `#add2all` if I want to add `test` to
`__all__`, although this way is not that elegant. And I may use `show_doc`
if I want to show the document which is hidden by `#exporti` before. I think
the feature of this flag `#exporti` is not simple enough.
And sometimes I found that the notebook itself is more readable than the
document generated from it.
Anyway, thanks for creating this cool tool.

**Comment by MartinKielhornPhilips** [comment]
I just stumbled on this issue as well. It is quite confusing that __all__ can be populated with random stuff picked up from string literals. But luckily I found this issue documented here.

In my case I had
`__all__ = [ ..., 'k']
`
For a while I was really confused where the 'k' variable comes from. It is also difficult to search.

But with this statement:

> 0 indent assignment statement variables will be put into __all__

I was able to easily home in on the issue with single grep:

```
$ grep '^k' python_api.py
k=1,2,...,8. The maximum for `lines` is 4096.
```
